python/tank/pipelineconfig.py

```diff
@@ -41,6 +42,12 @@ class PipelineConfiguration(object):
     to construct this object, do not create directly via the constructor.
     """
 
+    # A set of strings for the custom hook metrics that have already been
+    # logged. Certain custom hooks can be called many, many times
+    # (cache_location.get_path_cache_path). So we use this set of names to
+    # ensure we only log those metrics once.
+    __logged_custom_hook_metrics = set()
```

Comment:
maybe put a note in here to explain that we considered putting this in the singleton but it didn't make sense because things got quite complicated! I think if we start looking at adding similar logic in *other* places too, we really should consider some form of centralization of this, and then it's useful to have that kind of comment to explain why we didn't take that appraoch in the first place.